internal/circuit/redis.go

```
@@ -0,0 +1,193 @@
+
+// GetState returns the current circuit state, honouring the TTL-based
+// cooldown (Open state key expires → transitions to HalfOpen).
+func (s *RedisStore) GetState(ctx context.Context, provider string) (State, error) {
+	val, err := s.rdb.Get(ctx, s.stateKey(provider)).Result()
+	if err == redis.Nil {
+		return StateClosed, nil
+	}
+	if err != nil {
+		// On Redis errors, fail open (treat as closed) to avoid taking down the
+		// service because of a Redis blip.
+		return StateClosed, nil
```

golangci-lint (2.11.4)

[error] error is not nil (line 58) but it returns nil

(nilerr)